### ISSUE TITLE
Read messages keep changing status back to unread

### ISSUE THREAD

**Issue by golferguy** [thunderbird/thunderbird-android#1130]
### Expected behaviour

After I open and read an email, the status should remain unread and then sync with settings to poll every 15 minutes
### Actual behaviour

After I open and read an email, the status changes to read
But once the imap sync starts the status changes back to unread after polling
### Steps to reproduce

1.Launch K9 
2.View list of  emails and read one or more by tapping to open fully
3.Exit K9
4. Check K9 later and the previous read emails show as unread

If I read the messages on the server via outlook client on PC, the K9 client shows them as read messages, properly from that point forward
### Environment

K-9 Mail version:  5.008

Android version:
Marshmellow
LG G4

Account type (IMAP, POP3, WebDAV/Exchange):
IMAP settings but Activesync is used on servers

Note: This occurred about a week ago.
I exported settings
Deleted K9 from phone
Downloaded new version from Play store
Imported settings
K9 functioned perfectly for about a week but problem returned today.

**Comment by cketti** [comment]
Sounds like the "mark as read" commands either fail or are stuck in the local command queue and never sent to the server. Can you provide a debug log? https://github.com/k9mail/k-9/wiki/LoggingErrors

**Comment by golferguy** [comment]
Thank you for quick response. I have debug ready to submit but would like to remove password if it is stored in debug log as indicated in instructions.  But not sure I see it in txt file.  Are passwords in log file?

**Comment by golferguy** [comment]
Hello.
Would appreciate a response to ensuring password is removed before submitting log.  Also where is the upload  button to updoad a log?  I have captured the issue in the log

**Comment by cketti** [comment]
The log shouldn't contain any passwords unless you enabled the 'Log sensitive information' setting.

When composing a comment you will see the text "Attach files by dragging & dropping, selecting them, or pasting from the clipboard." below the input field. Drop your file there or click the "selecting them" link.

**Comment by golferguy** [comment]
I have attached the log.  The log should have captured reading two new emails, they then showed READ. I then had it sync with the server and then the emails showed UNREAD.  I then used ADB for the log file.
[k9-log.txt](https://github.com/k9mail/k-9/files/152155/k9-log.txt)

**Comment by philipwhiuk** [comment]
The log shows the same crypto / IO errors as the other issue. 
```
02-28 12:10:43.362 E/k9      (24475): com.fsck.k9.mail.MessagingException: Could not find UID for message based on Message-ID
02-28 12:10:43.362 E/k9      (24475): 	at com.fsck.k9.mail.store.ImapStore$ImapFolder.getUidFromMessageId(ImapStore.java:2105)
02-28 12:10:43.362 E/k9      (24475): 	at com.fsck.k9.mail.store.ImapStore$ImapFolder.appendMessages(ImapStore.java:2054)
02-28 12:10:43.362 E/k9      (24475): 	at com.fsck.k9.controller.MessagingController.processPendingAppend(MessagingController.java:2117)
02-28 12:10:43.362 E/k9      (24475): 	at com.fsck.k9.controller.MessagingController.processPendingCommandsSynchronous(MessagingController.java:1987)
02-28 12:10:43.362 E/k9      (24475): 	at com.fsck.k9.controller.MessagingController.access$1000(MessagingController.java:111)
02-28 12:10:43.362 E/k9      (24475): 	at com.fsck.k9.controller.MessagingController$12.run(MessagingController.java:1935)
02-28 12:10:43.362 E/k9      (24475): 	at com.fsck.k9.controller.MessagingController.run(MessagingController.java:435)
02-28 12:10:43.362 E/k9      (24475): 	at java.lang.Thread.run(Thread.java:818)
02-28 12:10:43.362 E/k9      (24475): Caused by: javax.net.ssl.SSLException: Read error: ssl=0x7f8cf6a600: I/O error during system call, Connection reset by peer
02-28 12:10:43.362 E/k9      (24475): 	at com.android.org.conscrypt.NativeCrypto.SSL_read(Native Method)
02-28 12:10:43.362 E/k9      (24475): 	at com.android.org.conscrypt.OpenSSLSocketImpl$SSLInputStream.read(OpenSSLSocketImpl.java:70)
02-28 12:10:43.362 E/k9      (24475): 	at java.io.InputStream.read(InputStream.java:162)
02-28 12:10:43.362 E/k9      (24475): 	at java.io.BufferedInputStream.fillbuf(BufferedInputStream.java:149)
02-28 12:10:43.362 E/k9      (24475): 	at java.io.BufferedInputStream.read(BufferedInputStream.java:234)
02-28 12:10:43.362 E/k9      (24475): 	at com.fsck.k9.mail.filter.PeekableInputStream.read(PeekableInputStream.java:25)
02-28 12:10:43.362 E/k9      (24475): 	at com.fsck.k9.mail.store.ImapResponseParser.readStringUntil(ImapResponseParser.java:349)
02-28 12:10:43.362 E/k9      (24475): 	at com.fsck.k9.mail.store.ImapResponseParser.parseResponseText(ImapResponseParser.java:118)
02-28 12:10:43.362 E/k9      (24475): 	at com.fsck.k9.mail.store.ImapResponseParser.readTokens(ImapResponseParser.java:73)
02-28 12:10:43.362 E/k9      (24475): 	at com.fsck.k9.mail.store.ImapResponseParser.readResponse(ImapResponseParser.java:45)
02-28 12:10:43.362 E/k9      (24475): 	at com.fsck.k9.mail.store.ImapResponseParser.readResponse(ImapResponseParser.java:29)
02-28 12:10:43.362 E/k9      (24475): 	at com.fsck.k9.mail.store.ImapStore$ImapConnection.readStatusResponse(ImapStore.java:2775)
02-28 12:10:43.362 E/k9      (24475): 	at com.fsck.k9.mail.store.ImapStore$ImapConnection.executeSimpleCommand(ImapStore.java:2925)
02-28 12:10:43.362 E/k9      (24475): 	at com.fsck.k9.mail.store.ImapStore$ImapConnection.executeSimpleCommand(ImapStore.java:2902)
02-28 12:10:43.362 E/k9      (24475): 	at com.fsck.k9.mail.store.ImapStore$ImapFolder.executeSimpleCommand(ImapStore.java:858)
02-28 12:10:43.362 E/k9      (24475): 	at com.fsck.k9.mail.store.ImapStore$ImapFolder.getUidFromMessageId(ImapStore.java:2094)
02-28 12:10:43.362 E/k9      (24475): 	... 7 more
```

Seems really odd that we'd be failing to get the UID like this.

**Comment by golferguy** [comment]
The phone k9 operation returned to normal yesterday.  I do not know why.  But there wasn't any phone reboots or app updates.   It suddenly started behaving correctly.

**Comment by cketti** [comment]
Unfortunately, the log doesn't help to narrow down the issue. But since the problem seems to be resolved for now I'm closing the issue.

**Comment by philipwhiuk** [comment]
From: #1193 

-----

Note: This occurred about a week ago.
I exported settings
Deleted K9 from phone
Downloaded new version from Play store
Imported settings
K9 functioned perfectly for about a week but problem returned today.

Problem returned on March 18, 2016

I noted in the K9mail-errors folder there are a number of emails showing with "IOException: readStringUntil(): end of stream reached"

And 
UnknownHostException: Unable to "resolve host "imap-mail.outlook.com":Unknown error"

---

The UnknownHostException indicates it couldn't contact Outlook (a temporary error and normal). The IOException is difficult to know whether it's temporary or not without more of the trace, but it's likely temporary.

Regardless, the way we handle synchronisation is an area we know needs work - the K9mail-errors folder is itself a particularly bad bit of UX. Networking issues should not result in an error being logged for the user - the user gets told their phone has no data connection in many other ways.

I don't see much value in reopening this issue unless we get a full stack trace indicating something we can definitely fix.

We don't support MS ActiveSync so that may well be part of the problem.

**Comment by rcoss** [comment]
I have the same behaviour as golferguy. I'm attaching an excerpt of  the debug log. Lots of errors:
"Error running command 'processPendingCommands'
08-23 16:14:08.751 E/k9      (10414): java.lang.IllegalArgumentException: the bind value at index 1 is null"
Complete log has too much sensible info.
@cketti Can send it in private.
K-9 Mail version: 5.010
[k9-log1.txt](https://github.com/k9mail/k-9/files/433529/k9-log1.txt)

**Comment by NuxRo** [comment]
This happens to me, too.

**Comment by grinapo** [comment]
Version: 5.700

It is not nice to necro-comment, but this is an open bug and the description fits well.
I have a pretty unhelpful log: 
```
Processing pending command 'com.fsck.k9.controller.MessagingControllerCommands$PendingMoveOrCopy@8d04de9'
```
results
```
Error running command 'processPendingCommands'
java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference
	at com.fsck.k9.mailstore.LocalFolder.<init>(LocalFolder.java:122)
	at com.fsck.k9.mailstore.LocalFolder.<init>(LocalFolder.java:112)
	at com.fsck.k9.mailstore.LocalFolder.<init>(LocalFolder.java:108)
	at com.fsck.k9.mailstore.LocalStore.getFolder(LocalStore.java:387)
	at com.fsck.k9.controller.MessagingController.processPendingMoveOrCopy(MessagingController.java:969)
	at com.fsck.k9.controller.MessagingController.processPendingMoveOrCopy(MessagingController.java:960)
	at com.fsck.k9.controller.MessagingControllerCommands$PendingMoveOrCopy.execute(MessagingControllerCommands.java:65)
	at com.fsck.k9.controller.MessagingController.processPendingCommandsSynchronous(MessagingController.java:822)
	at com.fsck.k9.controller.MessagingController$11.run(MessagingController.java:775)
	at com.fsck.k9.controller.MessagingController.runInBackground(MessagingController.java:220)
	at com.fsck.k9.controller.MessagingController.access$000(MessagingController.java:113)
	at com.fsck.k9.controller.MessagingController$1.run(MessagingController.java:163)
	at java.lang.Thread.run(Thread.java:919)
```

It seems that somehow K9 managed to open the mailbox in read-only. The problem is that in parallel there is another account on the same server (which is dovecot, btw) which works. Also in parallel there is another K9 (v5.600) which works on both accounts.

So far I do not see any problem in the IMAP packet dump, but I'll try to debug further within the server, yet I do not see any hope to tell why it fails.

**Comment by lars-vom-mars-x** [comment]
I have the same issue, suddenly for 2 days.
- The synchronised folders has the mark `IllegalArgumentException: b`.
see attached [k9-log-posteo.txt](https://github.com/k9mail/k-9/files/5193530/k9-log-posteo.txt)
- Other accounts like freenet works fine.
see attached [k9-log-freenet.txt](https://github.com/k9mail/k-9/files/5193534/k9-log-freenet.txt)
- **Why does the error only occur with posteo.de?**

Environment:
- K-9 Mail version: 5.717, installed 2020-06-20
- Android version: 9
- Device: FP3
- Account type: all IMAP

**Comment by cketti** [comment]
@lars-vom-mars-x: Updating to 5.718 should fix your issue.

**Comment by cketti** [comment]
Closing this issue. If you experience the same behavior please create a new issue.

**Comment by lars-vom-mars-x** [comment]
Thanks. Fixed. :-)

**Comment by sevenrock** [comment]
> Sounds like the "mark as read" commands either fail or are stuck in the local command queue and never sent to the server.

Having the same problem with k9 5.724 and an IMAP account. Accidently I tried to move mails to a mail folder which is a directory on the server and can therefore not store any mails. This move command got stuck in the local command queue:
https://gist.github.com/sevenrock/f307d73819455c55ccc30300220d0a0f
and blocks other commands, like changing the read-status of mails.

Is there a way to clear the local command queue without reinstalling k9?

**Comment by ja6** [comment]
I'm using 5.718 on Android 10 and have experienced the same issue.

> Is there a way to clear the local command queue without reinstalling k9?

Yes, via the Android settings you can clear the cache. This resolved the issue for me - at least temporarily.
